qiskit/dagcircuit/dagcircuit.py

```diff
@@ -94,6 +94,11 @@ def clbits(self):
         """Return a list of classical bits (as a list of Clbit instances)."""
         return [clbit for creg in self.cregs.values() for clbit in creg]
 
+    @property
+    def wires(self):
+        """Return a generator that returns wires in order."""
+        return sorted(self._wires, key=lambda bit: (bit.register.name, bit.index))
```

Comment:
if I create a circuit like this:
```
b_0: -----
b_1: -----
a_0: -----
a_1:  -----
```
I think this would cause the order to change to be like this:
```
a_0: -----
a_1: -----
b_0: -----
b_1:  -----
```

perhaps it should be a dict where the index associated with each one is saved.

Reply:
I was going based on @kdk's suggestion here. If we want to maintain insertion order as the order returned by `wires` that is what https://github.com/Qiskit/qiskit-terra/pull/3934/commits/4af3ce558f2285e522602b145ff76d8da72620e2 did. We can accomplish the same thing with an `OrderedDict` too if we don't want to use an external lib (which is what I asked yesterday).

Reply:
This could be fixed to preserve insertion order with:
```python
@property
def wires(self):
  return [ bit for reg in self.qregs + self.cregs for bit in reg ]
```
The problem here is that the DAGCircuit internally is pretty messy, and there's already a lot of redundant information so it's easy for important information to be lost, or outdated. 

Ideally, https://github.com/Qiskit/rfcs/pull/15 should set an interface for how users specify bit ordering, and then we can come up with the right internal data structures to preserve that, while maintaining performance.

Reply:
Ok, I updated this in: https://github.com/Qiskit/qiskit-terra/pull/3934/commits/0d357f189ce832fbd01a27f048f88417aed26bd4